core/src/main/java/org/jivesoftware/sparkimpl/profile/AvatarPanel.java

```diff
@@ -214,8 +209,16 @@ public void finished() {
                     return;
                 }
                 */
-                setAvatar(new ImageIcon(avatarImage));
-                avatarFile = selectedFile;
+                //convert BufferedImage to bytes
+                try {
+                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
Did you know that a ``ByteArrayOutputStream`` does not need to be closed, or did you forget and were lucky that it's not a problem? :) Try-with-resources is a good construct to have here, regardless.